src/Gridify/Builder/BaseQueryBuilder.cs

```diff
@@ -201,7 +202,8 @@ private static object AddIndexerNullCheck(IGMap<T> gMap, object query)
       }
 
       // handle case-Insensitive search
-      if (value is not null && valueExpression.IsCaseInsensitive
+      if (value is not null && (valueExpression.IsCaseInsensitive
```

Comment:
There is another branch I was working on, and on that branch, I removed (refactored) all the code related to case-insensitive search from this file. so it would be nice if you could revert all the changes to this file. so I can add my refactoring code to your branch.  (including the method signature change, it is ok if temporarily your test fails)